src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +192,23 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        # validate format of value for date variable
+        if action_type == LogicActionTypes.variable:
+            form_var = self.context["form_variables"].variables[variable]
+            action_value = data.get("action", {}).get("value")
+
+            if form_var.data_type == FormVariableDataTypes.date:
+                try:
+                    date.fromisoformat(action_value)
```

Comment:
try writing a test with `{"action": {"value": 42}}` and see what happens :)

-> this would be a good testcase for a hypothesis test with all JSON types, but do it on the serializer itself rather than the entire endpoint.